variables.container.tf

```
@@ -6,7 +6,8 @@ variable "blob_properties" {
     last_access_time_enabled      = optional(bool)
     versioning_enabled            = optional(bool, true)
     container_delete_retention_policy = optional(object({
-      days = optional(number, 7)
+      enabled = optional(bool, true)
+      days    = optional(number, 7)
 
     }), { days = 7 })
```

There is duplication here, the default should just be `{}`. It will pick up the `7` from the variable type definition.

```suggestion
    }), {})
```